tree/tree/src/TBranch.cxx

```diff
@@ -1538,8 +1541,7 @@ Int_t TBranch::GetBulkEntries(Long64_t entry, TBuffer &user_buf)
    user_buf.SetBufferOffset(bufbegin);
 
    if (fCurrentBasket == nullptr) {
-      R__ASSERT(fExtraBasket == nullptr && "fExtraBasket should have been set to nullptr by GetFreshBasket");
-      fExtraBasket = basket;
+      fExtraBaskets.push_back(basket);
```

Comment:
@@pcanal I was thinking that maybe it would make sense to simply `delete basket` if `fExtrabaskets` is already `!= nullptr`. We do not need to recycle more than one, no?

Reply:
When prefetch is on, it loads several baskets at a time (see line 1400 (as of this PR) in TBranch.cxx.  So for the same reason we kept the one basket before, it makes sense to keep all the baskets.  Peak memory use does not change and it reduces memory churn (and the associated threading bottleneck on the call of malloc).